src/libs/collect.c

```diff
@@ -715,6 +676,335 @@ static GtkTreeModel *_create_filtered_model(GtkTreeModel *model, dt_lib_collect_
   return filter;
 }
 
+static gboolean date_cal_leave(GtkWidget *widget, GdkEvent *event, dt_lib_collect_t *d)
+{
+  gtk_widget_hide(d->date_window);
+  return FALSE;
+}
+static void date_cal_select(GtkCalendar *calendar, dt_lib_collect_t *d)
+{
+  // we write the selected date in the correct entry
+  GtkEntry *entry = GTK_ENTRY(g_object_get_data(G_OBJECT(calendar), "current-entry"));
+  if(entry)
+  {
+    guint y, m, dd;
+    gtk_calendar_get_date(calendar, &y, &m, &dd);
+
+    // we adjust the time part depending of the operator
+    char dtxt[40] = { 0 };
+    switch(dt_bauhaus_combobox_get(d->num_op))
+    {
+      case 0:
+      case 5:
+        sprintf(dtxt, "%04d:%02d:%02d%%", y, m, dd);
```

Comment:
could you please not use `sprintf()` (at all, if possible), but use `snprintf()`
some time ago i have converted all/most sprintf to snprintf..